### PR TITLE
Design documentation added.

### PR DESCRIPTION
**Pull request by janosmurai**
## Type Of Change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Any modification in the test cases
- [x] Any modification in the documentation

### Checklist:

* [x] I have read and followed the [contribution guideline](https://github.com/axem-solutions/.github/blob/main/CONTRIBUTING.md).
* [x] I have checked to ensure there aren't other open Pull Requests for the same update/change.
* [ ] Every new modification in the production code has been tested.
* [x] New and existing test cases pass.
* [ ] 100% unit test coverage achieved for every production code modification I made.
* [x] I have made corresponding changes to the documentation.

## Related Issue
Closing: 
[DEM-223](https://axem.atlassian.net/browse/DEM-223)

## Description
Design documentation file added.

[DEM-223]: https://axem.atlassian.net/browse/DEM-223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+1. **Tool images** (`tool_images`): This resource allows you to retrieve both local and registry tool 
+images.
+2. **Container Engine** (`container_engine`): his resource covers various operations related to the 
+container engine, including:
+    - Pulling images
+    - Running containers
+    - Removing images
+    - Searching for images
+3. **Registries** (`registries`): This resource contains information about the available registries.
```
Comment (vezdeneszter) [inline docs/design.md:34]: Add "By this resource you can perform actions such as:"

File: docs/design.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+this resource.
+5. **Development Environment Catalogs** (`dev_env_catalogs`): This resource deals with the available 
+catalogs. You can perform the following actions:
+    - Adding or deleting catalogs
+    - Listing the Development Environments within a catalog
+    - Cloning a Development Environment descriptor from a catalog 
+
+## Core Components
+
+The dependency graph of the core modules is illustrated below:
```
Comment (vezdeneszter) [inline docs/design.md:49]: This should be a figure caption under the image

File: docs/wp-content/core_class_diagram.png
Comment (vezdeneszter) [inline docs/wp-content/core_class_diagram.png:1]: typo: Registries / "delete_regsitry"

Comment (vezdeneszter) [reply docs/wp-content/core_class_diagram.png:1]: typo: DevEnv / "supporte_tool..."

File: docs/design.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+allows for the creation of a unified interface for all internal resources.
+
+The relationships between classes in the core modules can be observed in the 
+[Core Class Diagram](wp-content/core_class_diagram.png).
+
+## Third-party Modules
+
+### Typer
+This module is used to create a framework for the CLI commands. You can add a new command by 
+extending the list of functions in the main.py file and applying the @typer_cli.command() decorator.
```
Comment (vezdeneszter) [inline docs/design.md:63]: @typer_cli.command() shall be formatted as code

File: docs/design.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+As shown in the diagram, the platform module depends on the rest of the resources. This concept 
+allows for the creation of a unified interface for all internal resources.
+
+The relationships between classes in the core modules can be observed in the 
+[Core Class Diagram](wp-content/core_class_diagram.png).
+
+## Third-party Modules
+
+### Typer
```
Comment (vezdeneszter) [inline docs/design.md:61]: shall be highlighted in bold. Apply the same on the other subtitles

File: docs/design.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    - Pulling an image from a registry
+4. **Local Development Environments** (`local_dev_envs`):  This resource provides a list of local 
+Development Environments. You can obtain information about a specific Development Environment using 
+this resource.
+5. **Development Environment Catalogs** (`dev_env_catalogs`): This resource deals with the available 
+catalogs. You can perform the following actions:
+    - Adding or deleting catalogs
+    - Listing the Development Environments within a catalog
+    - Cloning a Development Environment descriptor from a catalog 
+
```
Comment (vezdeneszter) [inline docs/design.md:46]: ADD CTA to XM discord channel : 

{% cta https://discord.com/invite/Nv6hSzXruK) %}  Join our Community:  Everything about Embedded Developer Tools and Developer Platform {% endcta %}

Comment (vezdeneszter) [reply docs/design.md:46]: please add this to the other pages as well

**Review by vezdeneszter — CHANGES_REQUESTED** [review]

File: docs/design.md (hunk 5 above)
Comment (janosmurai) [reply docs/design.md:47]: Will be fixed in [DEM-226](https://axem.atlassian.net/browse/DEM-226).

**Review by vezdeneszter — APPROVED** [review]

**Pull request merged**